src/services/donations/updateEntityCounters.js

```diff
@@ -147,75 +147,79 @@ const updateEntity = async (app, id, type) => {
   if (method === 'create' && donation.status === DonationStatus.PAID) {
-    app
-      .service('milestones')
-      .get(milestoneId)
-      .then(milestone => {
-        const { recipient } = milestone;
-        const { txHash, amount } = donation;
-        const { symbol, decimals } = donation.token;
-        const service = app.service('conversations');
-
-        conversationSem.take(async () => {
-          try {
-            const data = await service.find({
-              paginate: false,
-              query: {
-                milestoneId,
-                messageContext: 'payment',
-                txHash,
-                $limit: 1,
-              },
-            });
-
-            if (data.length > 0) {
-              const conversation = data[0];
-              const payments = conversation.payments || [];
-              const index = payments.findIndex(p => p.symbol === symbol);
+    const { txHash } = donation;
+    const events = (
+      await app.service('events').find({
+        query: {
+          transactionHash: donation.txHash,
+          event: 'Transfer',
+          status: { $ne: 'Processed' },
+        },
+      })
+    ).data;
+    if (events.length !== 0) {
+      logger.info(
+        'Dont create conversation when there is unProcessed Transfer events for a transactionHash',
+      );
+      return;
+    }
 
-              if (index !== -1) {
-                payments[index].amount = toBN(amount)
-                  .add(toBN(payments[index].amount))
-                  .toString();
-              } else {
-                payments.push({ symbol, amount, tokenDecimals: decimals });
-              }
+    try {
+      const milestone = await app.service('milestones').get(milestoneId);
+      const { recipient } = milestone;
+      const donations = (
+        await app.service('donations').find({
+          query: {
+            ownerTypeId: milestoneId,
+            status: DonationStatus.PAID,
+            txHash,
+          },
+        })
+      ).data;
+      const payments = [];
 
-              await service.patch(conversation._id, {
-                payments,
-              });
-            } else {
-              await service.create(
-                {
-                  milestoneId,
-                  messageContext: 'payment',
-                  txHash,
-                  payments: [
-                    {
-                      amount,
-                      symbol,
-                      tokenDecimals: decimals,
-                    },
-                  ],
-                  recipientAddress: recipient.address,
-                },
-                { performedByAddress: donation.actionTakerAddress },
-              );
-            }
-          } catch (e) {
-            logger.error('could not create conversation', e);
-          } finally {
-            conversationSem.leave();
-          }
+      // why our linter has problem with for..of ?
+      /* eslint-disable no-restricted-syntax */
```

Comment:
@aminlatifi 

Why our linter has problem with for..of,  is there a logical reason or it's just a legacy linter?